File: app/src/main/java/com/itba/runningMate/db/RunConverters.java
package com.itba.runningMate.db;

import androidx.room.TypeConverter;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.Date;
import java.util.List;

public class RunConverters {

    private static final Gson gson = new Gson();

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public static List<List<LatLng>> jsonToLatLngList(String route) {
        if (route == null) {
            return Collections.emptyList();
        }

        Type listType = new TypeToken<List<List<LatLng>>>() {
        }.getType();
        return gson.fromJson(route, listType);
    }

    @TypeConverter
    public static String LatLngListToJson(List<List<LatLng>> route) {
        if (route == null) {
            return null;
        }
        Type type = new TypeToken<List<List<LatLng>>>() {
        }.getType();
        return gson.toJson(route, type);
    }

}

File: app/src/main/java/com/itba/runningMate/db/RunConverters.kt
package com.itba.runningMate.db

import androidx.room.TypeConverter
import com.google.android.gms.maps.model.LatLng
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.util.*

object RunConverters {
    private val gson = Gson()

    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }

    @TypeConverter
    fun jsonToLatLngList(route: String?): List<List<LatLng>> {
        if (route == null) {
            return emptyList()
        }
        val listType = object : TypeToken<List<List<LatLng>>>() {}.type
        return gson.fromJson(route, listType)
    }

    @TypeConverter
    fun LatLngListToJson(route: List<List<LatLng>>?): String? {
        if (route == null) {
            return null
        }
        val type = object : TypeToken<List<List<LatLng>>>() {}.type
        return gson.toJson(route, type)
    }
}